src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+
+        core::SRange<const IGPUComputePipeline::SCreationParams> creationParamsRange(&creationParams,
+            &creationParams + 1);
+
+        if (createGPUComputePipelines_impl(_pipelineCache, creationParamsRange, &result))
+        {
+            return result;
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
-    bool createGPUComputePipelines_impl(IGPUPipelineCache* pipelineCache, core::SRange<const IGPUComputePipeline::SCreationParams> createInfos,
+    bool createGPUComputePipelines_impl(IGPUPipelineCache* pipelineCache,
+        core::SRange<const IGPUComputePipeline::SCreationParams> createInfos,
         core::smart_refctd_ptr<IGPUComputePipeline>* output) override
     {
-        return false;
+        constexpr uint32_t MAX_PIPELINE_COUNT = 100u;
+
+        assert(createInfos.size() <= MAX_PIPELINE_COUNT);
+
+        const IGPUComputePipeline::SCreationParams* creationParams = createInfos.begin();
+        for (size_t i = 0ull; i < createInfos.size(); ++i)
+        {
+            if ((creationParams[i].layout->getAPIType() != EAT_VULKAN) ||
+                (creationParams[i].shader->getAPIType() != EAT_VULKAN))
+            {
+                return false;
+            }
+        }
+
+        VkPipelineCache vk_pipelineCache = VK_NULL_HANDLE;
+        if (pipelineCache && pipelineCache->getAPIType() == EAT_VULKAN)
+            vk_pipelineCache = static_cast<const CVulkanPipelineCache*>(pipelineCache)->getInternalObject();
+
+        VkPipelineShaderStageCreateInfo vk_shaderStageCreateInfos[MAX_PIPELINE_COUNT];
+
+        VkComputePipelineCreateInfo vk_createInfos[MAX_PIPELINE_COUNT];
+        for (size_t i = 0ull; i < createInfos.size(); ++i)
+        {
+            const auto createInfo = createInfos.begin() + i;
+
+            vk_createInfos[i].sType = VK_STRUCTURE_TYPE_COMPUTE_PIPELINE_CREATE_INFO;
+            vk_createInfos[i].pNext = nullptr; // pNext must be either NULL or a pointer to a valid instance of VkPipelineCompilerControlCreateInfoAMD, VkPipelineCreationFeedbackCreateInfoEXT, or VkSubpassShadingPipelineCreateInfoHUAWEI
+            vk_createInfos[i].flags = static_cast<VkPipelineCreateFlags>(createInfo->flags);
+
+            if (createInfo->shader->getAPIType() != EAT_VULKAN)
+                continue;
+
+            const CVulkanSpecializedShader* specShader
+                = static_cast<const CVulkanSpecializedShader*>(createInfo->shader.get());
+
+            vk_shaderStageCreateInfos[i].sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+            vk_shaderStageCreateInfos[i].pNext = nullptr; // pNext must be NULL or a pointer to a valid instance of VkPipelineShaderStageRequiredSubgroupSizeCreateInfoEXT
+            vk_shaderStageCreateInfos[i].flags = 0; // currently there is no way to get this in the API https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkPipelineShaderStageCreateFlagBits.html
```

Comment:
we dont do subgroup control, so need to get them (for now)

Reply:
Resolved